### DISCUSSION TITLE
Using the API to buy a product

### DISCUSSION THREAD

**Discussion by stefancroft** [hendt/ebay-api#52] [Q&A]
Hey folks.

I'm working on a small app which can buy and sell a product on eBay. I have my dev account setup and I am able to list a product using the hendt api (the example in the docs works well). The problem is purchasing a product via the product ID. I can browse the item using:

`eBay.buy.browse.getItem('v1|110531293874|0').then(item => {
        console.log(item);
    }).catch(e => {
        console.log(e)
    });`

But then the getCheckoutSession always throws an authentication error. 
 message: 'Access denied',
        longMessage: 'Insufficient permissions to fulfill the request.'

Should headers be intercepted and changed to include the buy scope? Any help would be really appreciated.

Thank you

**Comment by dantio** [comment]
Hey, I think you need `https://api.ebay.com/oauth/api_scope/buy.order` in the scope.

```js
eBay.auth.oAuth2.setScope([
    'https://api.ebay.com/oauth/api_scope',
    'https://api.ebay.com/oauth/api_scope/buy.order'
]);

```
